redux-form/lib/FieldArray.d.ts

```diff
@@ -0,0 +1,214 @@
+import { Component } from "react";
+import { ComponentConstructor, Validator } from "../index";
+
+/**
+ * These are the props to give to `FieldArray`.
+ * Any others will be passed through to your component.
+ */
+interface BaseFieldArrayProps {
+    /**
+     * A string path, in dot-and-bracket notation, corresponding to a value
+     * in the form values. It may be as simple as 'firstName' or as complicated
+     * as contact.billing.address[2].phones[1].areaCode.
+     *
+     * Required but made optional so interface can be used on decorated components.
+     */
+    name?: string;
+
+    /**
+     * A Component or stateless function to render the field array.
+     *
+     * Required but made optional so interface can be used on decorated components.
+     */
+    component?: ComponentConstructor,
+    // component?: Component<P, any> | StatelessComponent<P>,
+
+    /**
+     * Allows you to to provide a field-level validation rule. The function will be given the
+     * current value of the field and all the other form values. If the field is valid, it
+     * should return `undefined`, if the field is invalid, it should return an error
+     * (usually, but not necessarily, a `String`).
+     */
+    validate?: Validator|Validator[];
+
+    /**
+     * Allows you to to provide a field-level warning rule. The function will be given the
+     * current value of the field and all the other form values. If the field needs a warning,
+     * it should return the warning (usually, but not necessarily, a `String`). If the field
+     * does not need a warning, it should return `undefined`.
+     */
+    warn?: Validator|Validator[];
+
+    /**
+     * If true, the rendered component will be available with the
+     * getRenderedComponent() method. Defaults to false. Cannot be used if your
+     * component is a stateless function component.
+     */
+    withRef?: boolean;
+
+    props?: Object;
+}
+
+/**
+ * Declare FieldArray as this interface to specify the generics.
+ */
+export interface GenericFieldArray<T, FieldCustomProps> extends Component<BaseFieldArrayProps & FieldCustomProps, {}> {
```

Comment:
Different base props, `BaseFieldArrayProps` vs `BaseFieldProps`.

Are you questioning the need for `GenericField` interface?